Transfer/Sources/CreateTransferPresenter.swift

```diff
@@ -0,0 +1,236 @@
+        if sourceToken != nil {
+            loadTransferMethods()
+        } else {
+            userRepository.getUser { [weak self] result in
+                guard let strongSelf = self else {
+                    return
+                }
+                switch result {
+                case .failure(let error):
+                    strongSelf.view.hideLoading()
+                    strongSelf.view.showError(error, { () -> Void in
+                        strongSelf.loadCreateTransfer()
+                    })
+
+                case .success(let user):
+                    strongSelf.sourceToken = user?.token
+                    strongSelf.loadTransferMethods()
+                }
+            }
+        }
+    }
+
+    private func loadTransferMethods() {
+        transferMethodRepository.listTransferMethods { [weak self] result in
+            guard let strongSelf = self else {
+                return
+            }
+            switch result {
+            case .failure(let error):
+                strongSelf.view.hideLoading()
+                strongSelf.view.showError(error, { () -> Void in
+                    strongSelf.loadCreateTransfer()
+                })
+
+            case .success(let result):
+                guard let firstActivatedTransferMetod = result?.data?.first
+                    else {
+                        strongSelf.selectedTransferMethod = nil
+                        strongSelf.view.hideLoading()
+                        return
+                }
+                if strongSelf.selectedTransferMethod == nil {
+                    strongSelf.selectedTransferMethod = firstActivatedTransferMetod
+                }
+                strongSelf.createInitialTransfer()
+            }
+        }
+    }
+
+    private func createInitialTransfer() {
+        let transfer = HyperwalletTransfer.Builder(clientTransferId: clientTransferId,
+                                                   sourceToken: sourceToken ?? "",
+                                                   destinationToken: selectedTransferMethod.token ?? "")
+            .destinationCurrency(selectedTransferMethod.transferMethodCurrency)
+            .build()
+
+        transferRepository.createTransfer(transfer) { [weak self] result in
+            guard let strongSelf = self else {
+                return
+            }
+            strongSelf.view.hideLoading()
+            switch result {
+            case .failure(let error):
+                strongSelf.view.showError(error, { () -> Void in
+                    strongSelf.loadCreateTransfer()
+                })
+
+            case .success(let transfer):
+                strongSelf.availableBalance = transfer?.destinationAmount
+                strongSelf.view.showCreateTransfer()
+            }
+        }
+    }
+
+    // MARK: - Create Transfer Button Tapped
+    func createTransfer() {
+        let transfer = HyperwalletTransfer.Builder(clientTransferId: clientTransferId,
```

Comment:
Add following condition before the builder
``` if transferAllFundsIsOn {
            amount = nil
        } ```

Reply:
added guard